### PR TITLE
Port of the ConnectionFromArraySlice method

### PR DESCRIPTION
**Pull request by solher**
This is the port of https://github.com/graphql/graphql-relay-js/commit/238d724a6c14bbda2aff180f839f27e1961930d7 for graphql-go.

Related:

- issue https://github.com/graphql-go/relay/issues/20

### CODE REVIEW

**Comment by sogko** [comment]
@solher Thanks for this PR! Looks good 👍🏻 Appreciate it very much =) 
I have a couple of minor comments that I'll add to your commit

**Comment by solher** [comment]
No problem !

File: array_connection.go [hunk 1]
```diff
@@ -4,80 +4,113 @@ import (
...
 	"errors"
 	"fmt"
-	"math"
 	"strconv"
 	"strings"
 )
 
 const PREFIX = "arrayconnection:"
 
+type ArraySliceMetaInfo struct {
```
Comment (sogko) [inline array_connection.go]: Would you mind helping to add `json` tags to the struct?

```go
type ArraySliceMetaInfo struct {
	SliceStart  int `json:"sliceStart"`
	ArrayLength int `json:"arrayLength"`
}
```

File: array_connection.go [hunk 2]
```diff
@@ -119,13 +151,35 @@ func CursorForObjectInConnection(data []interface{}, object interface{}) Connect
 	return offsetToCursor(offset)
 }
 
-func getOffset(cursor ConnectionCursor, defaultOffset int) int {
+func getOffsetWithDefault(cursor ConnectionCursor, defaultOffset int) int {
```
Comment (sogko) [inline array_connection.go]: I noticed that `getOffsetWithDefault()` function is exported in `relay-js`.
It makes sense to export this to give user access to offset value from a connection string, similar to `CursorToOffset()`

Related commit with additional comments:
https://github.com/graphql/graphql-relay-js/commit/8f0fc3b2e848d328a98fa871aac7e84e88ab01a3#diff-fdeefc1c54f1bbbe87f2c14ef091b5b9

Comment (solher) [inline array_connection.go]: Should we keep CursorToOffset exported then ?

Comment (sogko) [inline array_connection.go]: Yeah, let's keep both exported, we'll try to achieve parity with `relay-js` as best as we can

Comment (solher) [inline array_connection.go]: Let's also export the `offsetToCursor` function too then ;)

Comment (sogko) [inline array_connection.go]: sounds good 👍🏻

**Comment by sogko** [comment]
@solher Thanks again for helping out and contributing to the library, appreciate it very much 👍🏻

**Comment by michaelchemani** [comment]
I know that PR has been merged a long time ago but do you think you guys could provide an example of how to use `ConnectionFromSliceArray ()`? My goal is to also use `database offsets` and I cannot wrap my head around it. Thank you.

**Pull request merged**